Libraries/pbxspec/Sources/Manager.cpp

```diff
@@ -86,9 +86,13 @@ findSpecification(std::vector<std::string> const &domains, std::string const &id
 {
     typename T::vector vector = findSpecifications <T> (domains, type);
```

Comment:
The first point is done already: see the line just below that comment that uses `{ *specification->basedOnDomain(), AnyDomain() }` as the domains, with the explicitly specified domain coming first.

For your second point, I thought about that, but there's also the possibility of an indirect loop. That could be handled by keeping a set of seen (domain, identifier) pairs while recursively resolving the inheritance. Just not written yet.